### PR TITLE
Fixed integer parsing to cover the full 32-bit signed range

### PR DESCRIPTION
**Pull request by iboB**

### CODE REVIEW

**Review by chadaustin — COMMENTED** [review]
Thanks! This looks good to me. Some small comments. :)

File: include/sajson.h [hunk 1]
```diff
@@ -2223,7 +2228,7 @@ class parser {
             if (try_double) {
                 d = -d;
             } else {
-                i = -i;
+                u = 0u - u;
```
[the remark this answers is not in the document]
Comment (iboB) [reply include/sajson.h:2231]: Yes, a (pedantic) warning for negating an unsigned number

File: include/sajson.h [hunk 2]
```diff
@@ -2056,6 +2056,9 @@ class parser {
     std::pair<char*, internal::tag> parse_number(char* p) {
         using internal::tag;
 
+        static constexpr unsigned RISKY = unsigned(INT_MAX) / 10;
+        unsigned max_digit_after_risky = unsigned(INT_MAX) % 10;
```
[the remark this answers is not in the document]
Comment (iboB) [reply include/sajson.h:2060]: The old code was using the macros, so I just kept the style. But, yeah, the casts are not really needed

**Comment by chadaustin** [comment]
Thank you!

**Pull request merged**